### DISCUSSION TITLE
How to Get Longer Voice Generate Duration

### DISCUSSION THREAD

**Discussion by Zavnok** [neonbjb/tortoise-tts#260] [General]
The last time I generated a `generated.wav` file it was only 23 seconds long and it didn't finish my input text sentence.

How can I get the fully generated output file?

**Comment by Zavnok** [comment]
Since no one has answered (at least yet), [here's what I have found](https://youtube.com/watch?v=FN3yxL0Rr0c&t=219s).

Just make sure to declare a `CUSTOM_VOICE_NAME` if you are skipping the **Upload your audios samples here** section.

It works for me now, but I'm open for suggestions.

**Reply by FurkanGozukara** [reply]
How much training dataset is good you think

Is longer better? I have hours of good quality speech

**Reply by Zavnok** [reply]
I have no idea about that. I use the default speeches included [here](https://github.com/jnordberg/tortoise-tts/tree/main/tortoise/voices).

**Reply by FurkanGozukara** [reply]
> I have no idea about that. I use the default speeches included [here](https://github.com/jnordberg/tortoise-tts/tree/main/tortoise/voices).

Thank you so much for answer. Do you know anyway I can give a subtitle srt or vtt file to give and generate entire speech?

**Reply by Zavnok** [reply]
1. Open the `srt` file with a text editor program and remove all the timestamps in it to make it look like a continuous speech. Here's an example:

Before editing the `srt` file:
```
1
00:05:00,400 --> 00:05:15,300
This is an example of
the first sentence.

2
00:05:16,400 --> 00:05:25,300
This is an example of
the second sentence.
```

After editing the `srt` file:
```
This is an example of the first sentence. This is an example of the second sentence.
```

2. Then save it as a `speech.txt` file.
3. Follow [this video](https://www.youtube.com/watch?v=FN3yxL0Rr0c) or use [this colab](https://colab.research.google.com/drive/1g_CssJK34kwRi7VRtFd73WvTLq9UbnZT?usp=sharing) to generate the entire speech using the above `speech.txt` file.

**Reply by FurkanGozukara** [reply]
@hashnimo  ye but that is not a solution. i need synching the voice

**Comment by shamoon-zafar** [comment]
Here is the detailed guide on [How to Generate Longer Speech with Tortoise TTS](https://www.aicolab.xyz/2023/12/generate-longer-speech-with-tortoise-tts.html)
